cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -442,30 +444,37 @@ fn send_weth_asset_from_asset_hub_to_ethereum() {
 				],
 			)),
 			fun: Fungible(WETH_AMOUNT),
-		}];
-		let multi_assets = VersionedAssets::V4(Assets::from(assets));
+		};
+		let assets = vec![fee_asset.clone(), weth_asset.clone()];
 
-		let destination = VersionedLocation::V4(Location::new(
-			2,
-			[GlobalConsensus(Ethereum { chain_id: CHAIN_ID })],
-		));
+		let destination = Location::new(2, [GlobalConsensus(Ethereum { chain_id: CHAIN_ID })]);
 
-		let beneficiary = VersionedLocation::V4(Location::new(
+		let beneficiary = Location::new(
 			0,
 			[AccountKey20 { network: None, key: ETHEREUM_DESTINATION_ADDRESS.into() }],
-		));
+		);
 
+		let xcm_on_bh = Xcm(vec![
+			BuyExecution { fees: fee_asset.clone(), weight_limit: Unlimited },
+			DepositAsset { assets: Wild(AllCounted(2)), beneficiary },
+		]);
+
+		let xcms = VersionedXcm::from(Xcm(vec![
+			WithdrawAsset(assets.clone().into()),
+			SetFeesMode { jit_withdraw: true },
```

Comment:
Why `SetFeesMode` instead of `BuyExecution`? I think the user should have an opportunity to pass the necessary fee amounts after estimating fees using the XCM dry run extrinsic.

Also they want to deprecate `SetFeesMode` in https://github.com/polkadot-fellows/xcm-format/pull/57.

Reply:
The fee asset(DOT) is already included in `WithdrawAsset(assets.clone().into())`, in this test case actually we transfer 2 kinds of assets to Ethereum(DOT and WETH).

Reply:
Can't the executor take fees from the holding register? DOT has already been added to the holding register after the `WithdrawAsset` instruction

Reply:
We can but I don't see benefit for adding an extra `BuyExecution` if `SetFeesMode` can work here.

Btw: https://github.com/polkadot-fellows/xcm-format/pull/57 is closed so it won't impact and I expect more on https://github.com/polkadot-fellows/RFCs/pull/105 which is another story.